File: source/frontend/src/app/models/projects/catastrophe-type.ts
/**
 * Available types of catastrophes for projects
 *
 * It's not expected to add or remove types that often.
 * Therefore I decided to simple use enum instead of database record.
 * Should this fact change (don't think it will), it would be wiser to pull it up to database as an application entity,
 * and retrieve it from there.
 *
 */
export enum CatastropheType {
  DROUGHT = 'drought',
  HIGH_TEMPERATURES = 'high_temperatures',
  WIND = 'wind',
  HEAVY_RAINFALL = 'heavy_rainfall',
  FLOODING = 'flooding',
  FOOD_AND_WATER_OUTAGE = 'food_and_water_outage',
  BIOTIC_EMERGENCY = 'biotic_emergency',
  CHEMICAL_EMERGENCY = 'chemical_emergency',
  ENERGY_OUTAGE = 'energy_outage',
  MIGRATION = 'migration',
  CRIMINALITY = 'criminality',
  OTHER = 'other',
}

File: source/backend/src/main/kotlin/cz/opendatalab/egidio/backend/business/entities/project/CatastropheType.kt
package cz.opendatalab.egidio.backend.business.entities.project

import com.fasterxml.jackson.annotation.JsonValue

/**
 * Available types of catastrophes
 */
enum class CatastropheType(@field:JsonValue() val value: String) {
    DROUGHT(value = "drought"),
    HIGH_TEMPERATURES(value = "high_temperatures"),
    WIND(value = "wind"),
    HEAVY_RAINFALL(value = "heavy_rainfall"),
    FLOODING(value = "flooding"),
    FOOD_AND_WATER_OUTAGE(value = "food_and_water_outage"),
    BIOTIC_EMERGENCY(value = "biotic_emergency"),
    CHEMICAL_EMERGENCY(value = "chemical_emergency"),
    ENERGY_OUTAGE(value = "energy_outage"),
    MIGRATION(value = "migration"),
    CRIMINALITY(value = "criminality"),
    OTHER(value = "other"),
}